unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Utils/ProtoConvertUtils.cs

```diff
@@ -168,5 +169,49 @@ public static DCLImageScaleMode ToDCLImageScaleMode(this BackgroundTextureMode t
                        _ => DCLImageScaleMode.STRETCH
                    };
         }
+
+        public static int SDKCollisionMaskToUnityLayerMask(uint sdkLayerMask)
+        {
+            bool hasPhysicsLayer = false;
+            bool hasPointerLayer = false;
+            int unityLayerMask = new LayerMask();
+
+            if (IsInLayerMask(sdkLayerMask, (int)ColliderLayer.ClPointer))
+            {
+                unityLayerMask |= (1 << PhysicsLayers.onPointerEventLayer);
+                hasPointerLayer = true;
+            }
+
+            if (IsInLayerMask(sdkLayerMask, (int)ColliderLayer.ClPhysics))
+            {
+                unityLayerMask |= (1 << PhysicsLayers.characterOnlyLayer);
+                hasPhysicsLayer = true;
+            }
+
+            if(hasPointerLayer && hasPhysicsLayer)
+                unityLayerMask |= (1 << PhysicsLayers.defaultLayer);
+
+            if (LayerMaskHasAnySDKCustomLayer(sdkLayerMask))
+            {
+                unityLayerMask |= (1 << PhysicsLayers.sdkCustomLayer);
+            }
+
+            return unityLayerMask;
+        }
```

Comment:
this function got me confused:
1- why are we creating a `new LayerMask()` if function returns an `int`?
2- is this function only used to calculate the layer for the raycast? 
maybe we should move it to where it is used instead of here and rename it for something like `GetRaycastTargetLayerMask` or something like that. WDYT?

Reply:
Done